/*========================== begin_copyright_notice ============================

Copyright (C) 2021 Intel Corporation

SPDX-License-Identifier: MIT

============================= end_copyright_notice ===========================*/

syntax = "proto3";

package IGC_METRICS;

message InstrStats {
  enum SIMDType {
    SIMD1 = 0;
    SIMD4 = 1;
    SIMD8 = 2;
    SIMD16 = 3;
    SIMD32 = 4;
  }

  repeated SIMDType simd_used = 1;
  int32 countLoads = 2;
  int32 countStores = 3;
  int32 countCoalescedAccess = 4;
  int32 countShufflePermutes = 5;
  int32 countArithmeticInst = 6;
  int32 countEmulatedInst = 7;
  int32 countTranscendentalFunc = 8;
}